### PR TITLE
User class verification

### PR DESCRIPTION
**Pull request by Aliblas**
Passing the Google ID token to the back end for every user operation seems annoying, so I'm hoping to investigate alternatives soon.
Also considering @JSType to replace the operations in UserActionServlet, for the sake of elegance.

### CODE REVIEW

**Review by isamitre — APPROVED** [review]

File: src/main/java/com/google/sps/data/User.java [hunk 1]
```diff
@@ -55,33 +57,47 @@
...
+   * throws SecurityException on failure, import from java.lang.SecurityException.
+   * @Param idTokenString is the Google-user ID Token provided buy user-auth.js/getIdToken.
+   * Token should be passed as URL Fetch argument from front end.
    */
-  public User() {
-    String testId = TEST_ID;
-    Key userKey = KeyFactory.createKey("User", testId);
+  public User(String idTokenString) throws SecurityException {
+    GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier
+    .Builder(UrlFetchTransport.getDefaultInstance(), new JacksonFactory())
```
Comment (AlexanderEllis) [inline src/main/java/com/google/sps/data/User.java:67]: Nit: Indent 4 spaces for a line continuation: https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent

File: src/main/java/com/google/sps/data/User.java [hunk 2]
```diff
@@ -55,33 +57,47 @@
   private Entity entity;
 
   /**
-   * This creates a standard universal user instance regardless of login status.
-   * This method will be changed to instantiate distinct user object.
+   * Verifies user and creates a User instance for accessing and adding user data.
+   * throws SecurityException on failure, import from java.lang.SecurityException.
```
Comment (AlexanderEllis) [inline src/main/java/com/google/sps/data/User.java:61]: Super nit: capitalize T in `throws`

File: src/main/java/com/google/sps/data/User.java [hunk 3]
```diff
@@ -55,33 +57,47 @@
...
+    } catch(IOException | GeneralSecurityException e) {
+      throw new SecurityException("Failed to verify Google token", e);
+    }
+
+    payload = idToken.getPayload();
+
+    Key userKey = KeyFactory.createKey("User", getId());
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     try {
       entity = datastore.get(userKey);
```
Comment (AlexanderEllis) [inline src/main/java/com/google/sps/data/User.java:83]: Are you doing anything with the entity here, or is this basically a check to see if the user already exists? If you wouldn't mind adding a comment here that'd be great.

**Review by AlexanderEllis — APPROVED** [review]
LGTM with a few minor comments

**Pull request merged**